Proper reference to callback.

kuldeepdhaka opened the pull request:
even if the callback is NULL, set it.
note: not modifying EP0 callback
fixes issue #448 and #449

karlp commented:
I don't understand the commit comment here. Even after reading the linked issues.  For commits, the commit comment itself has to have the relevant information, not just links to offsite.  I don't see how the code matches this comment here.

kuldeepdhaka commented:
if callback is NULL, endpoint callback array is not updated. (previous reference is used)
if ep0 is being configured, endpoint callback array is update. (not required, as it is done in other parts)
removed check for callback NULL, and added check to only update if the ep is not ep0.
though the change look small, but fixes two bugs

kuldeepdhaka commented:
closed in favour of #508 (issue #448 and #449 are automatically being fixed)
